### PR TITLE
Separate tasks in each job of circleci

### PR DESCRIPTION
**Pull request by kan-bayashi**
This PR separates tasks in each job of circleci to check the bottleneck part. (#1042)

### CODE REVIEW

File: ci/test_python.sh [hunk 1]
```diff
@@ -10,4 +10,4 @@ flake8 espnet test utils doc;
 autopep8 -r espnet test utils --global-config .pep8 --diff --max-line-length 120 | tee check_autopep8
 test ! -s check_autopep8
 
-LD_LIBRARY_PATH="${LD_LIBRARY_PATH:-}:$(pwd)/chainer_ctc/ext/warp-ctc/build" pytest --cov=espnet
+LD_LIBRARY_PATH="${LD_LIBRARY_PATH:-}:$(pwd)/chainer_ctc/ext/warp-ctc/build" pytest --cov=espnet --durations=0
```
Comment (ShigekiKarita) [inline ci/test_python.sh]: How about moving these options to `setup.cfg`  like this
```
[tool:pytest]
addopts = --verbose --cov=espnet --durations=0
```
Maybe putting options in `setup.cfg` and `ci/test_python.sh` is hard to notice.

**Comment by ShigekiKarita** [comment]
According to Travis https://travis-ci.org/espnet/espnet/jobs/567469052, we have a lot of  slower tests in ASR and MT. Maybe tests over 3.0 sec are painful for me. Should we introduce [pytest-timeout](https://pypi.org/project/pytest-timeout/)? @kan-bayashi How do you think?
```

============================ slowest test durations ============================
16.82s call     test/test_e2e_asr_transformer.py::test_transformer_synth[chainer]
10.66s call     test/test_e2e_asr.py::test_sortagrad_trainable_with_batch_frames[chainer]
8.45s call     test/test_e2e_asr_transformer.py::test_transformer_synth[pytorch]
7.01s call     test/test_e2e_asr.py::test_sortagrad_trainable_with_batch_bins[chainer]
6.92s call     test/test_e2e_mt.py::test_sortagrad_trainable_with_batch_frames[pytorch]
4.90s call     test/test_e2e_compatibility.py::test_downloaded_asr_model_decodable[espnet.nets.chainer_backend.e2e_asr-download_info1]
4.62s call     test/test_e2e_asr.py::test_sortagrad_trainable[chainer]
4.37s call     test/test_e2e_asr.py::test_loss_and_ctc_grad[blstmp]
4.27s call     test/test_recog.py::test_recognition_results_with_lm[bgrup-gru-espnet.nets.chainer_backend.e2e_asr-4]
4.04s call     test/test_recog.py::test_recognition_results[bgrup-gru-espnet.nets.chainer_backend.e2e_asr-4]
3.88s call     test/test_e2e_asr.py::test_model_trainable_and_decodable[espnet.nets.chainer_backend.e2e_asr-model_dict9]
3.68s call     test/test_e2e_mt.py::test_sortagrad_trainable_with_batch_bins[pytorch]
3.55s call     test/test_recog.py::test_recognition_results_with_lm[blstmp-lstm-espnet.nets.chainer_backend.e2e_asr-0]
3.52s call     test/test_e2e_asr.py::test_model_trainable_and_decodable[espnet.nets.chainer_backend.e2e_asr-model_dict10]
3.40s call     test/test_e2e_asr.py::test_model_trainable_and_decodable[espnet.nets.chainer_backend.e2e_asr-model_dict7]
3.28s call     test/test_recog.py::test_recognition_results[blstmp-lstm-espnet.nets.chainer_backend.e2e_asr-0]
3.26s call     test/test_recog.py::test_recognition_results_with_lm[vggbgrup-gru-espnet.nets.chainer_backend.e2e_asr-6]
3.10s call     test/test_e2e_asr.py::test_loss_and_ctc_grad[vggblstmp]
3.02s call     test/test_recog.py::test_recognition_results[vggbgrup-gru-espnet.nets.chainer_backend.e2e_asr-6]
2.99s call     test/test_e2e_compatibility.py::test_downloaded_asr_model_decodable[espnet.nets.pytorch_backend.e2e_asr-download_info0]
2.98s call     test/test_recog.py::test_recognition_results_with_lm[vggblstmp-lstm-espnet.nets.chainer_backend.e2e_asr-2]
2.95s call     test/test_e2e_asr.py::test_model_trainable_and_decodable[espnet.nets.chainer_backend.e2e_asr-model_dict8]
2.73s call     test/test_recog.py::test_recognition_results[vggblstmp-lstm-espnet.nets.chainer_backend.e2e_asr-2]
2.70s call     test/test_e2e_asr.py::test_chainer_ctc_type
2.43s call     test/test_e2e_asr.py::test_model_trainable_and_decodable[espnet.nets.chainer_backend.e2e_asr-model_dict1]
2.38s call     test/test_e2e_asr.py::test_window_streaming_e2e_encoder_and_ctc_with_offline_attention
2.30s call     test/test_e2e_mt.py::test_context_residual[pytorch]
2.26s call     test/test_e2e_asr.py::test_mtl_loss[blstmp]
2.07s call     test/test_e2e_asr.py::test_model_trainable_and_decodable[espnet.nets.chainer_backend.e2e_asr-model_dict0]
1.98s call     test/test_e2e_asr.py::test_model_trainable_and_decodable[espnet.nets.pytorch_backend.e2e_asr-model_dict35]
1.92s call     test/test_e2e_asr.py::test_model_trainable_and_decodable[espnet.nets.chainer_backend.e2e_asr-model_dict5]
1.90s call     test/test_e2e_asr.py::test_model_trainable_and_decodable[espnet.nets.pytorch_backend.e2e_asr-model_dict36]
1.88s call     test/test_e2e_asr.py::test_sortagrad_trainable_with_batch_frames[pytorch]
1.88s call     test/test_e2e_mt.py::test_model_trainable_and_decodable[espnet.nets.pytorch_backend.e2e_mt-blstm-multi_head_add-lstm]
1.86s call     test/test_e2e_asr.py::test_model_trainable_and_decodable[espnet.nets.chainer_backend.e2e_asr-model_dict20]
1.82s call     test/test_e2e_asr.py::test_model_trainable_and_decodable[espnet.nets.chainer_backend.e2e_asr-model_dict15]
1.80s call     test/test_e2e_asr.py::test_model_trainable_and_decodable[espnet.nets.chainer_backend.e2e_asr-model_dict17]
1.80s call     test/test_e2e_asr.py::test_model_trainable_and_decodable[espnet.nets.chainer_backend.e2e_asr-model_dict16]
1.79s call     test/test_e2e_asr.py::test_model_trainable_and_decodable[espnet.nets.chainer_backend.e2e_asr-model_dict12]
1.76s call     test/test_e2e_asr.py::test_model_trainable_and_decodable[espnet.nets.chainer_backend.e2e_asr-model_dict28]
1.76s call     test/test_e2e_asr.py::test_model_trainable_and_decodable[espnet.nets.chainer_backend.e2e_asr-model_dict26]
1.75s call     test/test_e2e_asr.py::test_model_trainable_and_decodable[espnet.nets.chainer_backend.e2e_asr-model_dict27]
1.75s call     test/test_e2e_mt.py::test_sortagrad_trainable[pytorch]
1.75s call     test/test_e2e_asr.py::test_model_trainable_and_decodable[espnet.nets.chainer_backend.e2e_asr-model_dict25]
1.75s call     test/test_e2e_asr.py::test_model_trainable_and_decodable[espnet.nets.chainer_backend.e2e_asr-model_dict22]
1.74s call     test/test_e2e_asr.py::test_model_trainable_and_decodable[espnet.nets.chainer_backend.e2e_asr-model_dict21]
1.74s call     test/test_e2e_asr.py::test_model_trainable_and_decodable[espnet.nets.chainer_backend.e2e_asr-model_dict24]
1.74s call     test/test_e2e_asr.py::test_model_trainable_and_decodable[espnet.nets.chainer_backend.e2e_asr-model_dict18]
1.73s call     test/test_e2e_asr.py::test_model_trainable_and_decodable[espnet.nets.chainer_backend.e2e_asr-model_dict2]
1.71s call     test/test_e2e_asr.py::test_model_trainable_and_decodable[espnet.nets.chainer_backend.e2e_asr-model_dict23]
1.67s call     test/test_e2e_asr.py::test_model_trainable_and_decodable[espnet.nets.chainer_backend.e2e_asr-model_dict19]
1.67s call     test/test_e2e_asr.py::test_model_trainable_and_decodable[espnet.nets.chainer_backend.e2e_asr-model_dict4]
1.66s call     test/test_e2e_asr.py::test_model_trainable_and_decodable[espnet.nets.chainer_backend.e2e_asr-model_dict6]
1.60s call     test/test_e2e_asr.py::test_mtl_loss[vggblstmp]
1.56s call     test/test_e2e_asr.py::test_model_trainable_and_decodable[espnet.nets.chainer_backend.e2e_asr-model_dict3]
1.45s call     test/test_e2e_mt.py::test_model_trainable_and_decodable[espnet.nets.pytorch_backend.e2e_mt-blstm-multi_head_dot-lstm]
1.33s call     test/test_e2e_asr.py::test_model_trainable_and_decodable[espnet.nets.chainer_backend.e2e_asr-model_dict13]
1.32s call     test/test_e2e_asr.py::test_model_trainable_and_decodable[espnet.nets.chainer_backend.e2e_asr-model_dict11]
1.31s call     test/test_e2e_asr.py::test_sortagrad_trainable_with_batch_bins[pytorch]
1.30s call     test/test_e2e_mt.py::test_model_trainable_and_decodable[espnet.nets.pytorch_backend.e2e_mt-blstm-coverage-lstm]
1.26s call     test/test_e2e_asr.py::test_model_trainable_and_decodable[espnet.nets.chainer_backend.e2e_asr-model_dict14]
1.09s call     test/test_e2e_mt.py::test_model_trainable_and_decodable[espnet.nets.pytorch_backend.e2e_mt-blstm-add-lstm0]
1.08s call     test/test_e2e_mt.py::test_model_trainable_and_decodable[espnet.nets.pytorch_backend.e2e_mt-blstmp-add-lstm]
1.03s call     test/test_e2e_mt.py::test_model_trainable_and_decodable[espnet.nets.pytorch_backend.e2e_mt-blstm-add-lstm1]
1.01s call     test/test_e2e_mt.py::test_model_trainable_and_decodable[espnet.nets.pytorch_backend.e2e_mt-bgru-add-lstm]
```

File: .circleci/config.yml [hunk 2]
```diff
@@ -87,24 +123,42 @@ jobs:
...
+      # - run:
+      #     name: install kaldi
+      #     command: |
+      #       # TODO(karita) centos7 featbin binary
+      #       PATH="/opt/rh/devtoolset-7/root/usr/bin:${PATH:-}"
+      #       ./ci/install_kaldi.sh
+      - run:
+          name: test integration
+          command: |
+            ./ci/test_integration.sh
```
Comment (ShigekiKarita) [inline .circleci/config.yml]: You cannot run this in CentOS7 because I haven't prepared kaldi featbin for CentOS7 (old libc distribution).

**Comment by kan-bayashi** [comment]
@ShigekiKarita I agree with your idea. We should care about the runtime of each test.
BTW, some tests takes too much time in circleci. 
Maybe some problems were happened...
https://circleci.com/gh/espnet/espnet/4459

**Comment by kan-bayashi** [comment]
OMG
https://circleci.com/gh/espnet/espnet/4509
```
============================ slowest test durations ============================
352.25s call     test/test_e2e_mt.py::test_sortagrad_trainable_with_batch_frames[pytorch]
94.16s call     test/test_e2e_mt.py::test_sortagrad_trainable_with_batch_bins[pytorch]
81.24s call     test/test_e2e_asr.py::test_sortagrad_trainable_with_batch_frames[pytorch]
75.24s call     test/test_e2e_mt.py::test_context_residual[pytorch]
60.15s call     test/test_e2e_mt.py::test_sortagrad_trainable[pytorch]
42.72s call     test/test_e2e_asr.py::test_sortagrad_trainable_with_batch_bins[pytorch]
33.10s call     test/test_e2e_asr_transformer.py::test_transformer_synth[pytorch]
25.71s call     test/test_e2e_mt.py::test_model_trainable_and_decodable[espnet.nets.pytorch_backend.e2e_mt-blstm-multi_head_add-lstm]
22.60s call     test/test_e2e_asr.py::test_sortagrad_trainable[pytorch]
22.60s call     test/test_e2e_mt.py::test_model_trainable_and_decodable[espnet.nets.pytorch_backend.e2e_mt-blstm-add-lstm1]
21.12s call     test/test_e2e_mt.py::test_loss[blstm]
19.45s call     test/test_e2e_mt.py::test_model_trainable_and_decodable[espnet.nets.pytorch_backend.e2e_mt-blstm-multi_head_dot-lstm]
16.71s call     test/test_e2e_mt.py::test_model_trainable_and_decodable[espnet.nets.pytorch_backend.e2e_mt-blstmp-add-lstm]
16.09s call     test/test_e2e_asr_transformer.py::test_transformer_synth[chainer]
14.75s call     test/test_e2e_mt.py::test_model_trainable_and_decodable[espnet.nets.pytorch_backend.e2e_mt-blstm-add-lstm0]
14.74s call     test/test_e2e_mt.py::test_model_trainable_and_decodable[espnet.nets.pytorch_backend.e2e_mt-blstm-noatt-lstm]
14.48s call     test/test_e2e_mt.py::test_model_trainable_and_decodable[espnet.nets.pytorch_backend.e2e_mt-blstm-coverage-lstm]
13.41s call     test/test_e2e_asr.py::test_loss_and_ctc_grad[blstmp]
12.23s call     test/test_e2e_asr.py::test_gradient_noise_injection[pytorch]
11.67s call     test/test_e2e_asr.py::test_sortagrad_trainable_with_batch_frames[chainer]
11.48s call     test/test_e2e_asr.py::test_loss_and_ctc_grad[vggblstmp]
11.18s call     test/test_e2e_mt.py::test_model_trainable_and_decodable[espnet.nets.pytorch_backend.e2e_mt-bgru-add-lstm]
10.07s call     test/test_e2e_mt.py::test_model_trainable_and_decodable[espnet.nets.pytorch_backend.e2e_mt-lstmp-add-lstm]
8.16s call     test/test_e2e_mt.py::test_model_trainable_and_decodable[espnet.nets.pytorch_backend.e2e_mt-blstm-dot-lstm]
7.56s call     test/test_e2e_mt.py::test_model_trainable_and_decodable[espnet.nets.pytorch_backend.e2e_mt-bgrup-add-lstm]
7.29s call     test/test_e2e_asr.py::test_mtl_loss[blstmp]
6.81s call     test/test_e2e_asr.py::test_sortagrad_trainable_with_batch_bins[chainer]
6.58s call     test/test_e2e_asr.py::test_model_trainable_and_decodable[espnet.nets.pytorch_backend.e2e_asr-model_dict36]
6.56s call     test/test_e2e_asr.py::test_model_trainable_and_decodable[espnet.nets.pytorch_backend.e2e_asr-model_dict34]
6.11s call     test/test_e2e_mt.py::test_model_trainable_and_decodable[espnet.nets.pytorch_backend.e2e_mt-grup-add-lstm]
5.95s call     test/test_e2e_asr.py::test_model_trainable_and_decodable[espnet.nets.pytorch_backend.e2e_asr-model_dict35]
5.82s call     test/test_e2e_asr.py::test_model_trainable_and_decodable[espnet.nets.pytorch_backend.e2e_asr-model_dict30]
5.37s call     test/test_e2e_asr.py::test_model_trainable_and_decodable[espnet.nets.pytorch_backend.e2e_asr-model_dict54]
```

**Comment by ShigekiKarita** [comment]
LOL. What is wrong with conda (or maybe docker)?

**Comment by ShigekiKarita** [comment]
Maybe we should revert `USE_CONDA=true` at least one case in Travis for searching what is wrong.

**Pull request merged**